### DISCUSSION TITLE
How to show custom list columns (not property columns)

### DISCUSSION THREAD

**Discussion by RamBalani** [microsoft-search/pnp-modern-search#1974] [Q&A]
**PnP Search Box - and PnP Search Result  Web parts in Modern page are working with Custom List BUT Search Results page is not showing Allitems View Columns  (* Default and one and only one view) 
HOW DO I ADD THE ACTUAL COLUMNS ON TH ELIST BEING SEARCHED? THE SEARCH SEEMS TO BE OKAY -I verified the search hit results with keywords via a Classic stand-alone version of the same exact LIST.

--------------
[PnP Search BOx-Search Result Workign with Custom List-Not showing ALLitems View Columns.pdf](https://github.com/microsoft-search/pnp-modern-search/files/8364015/PnP.Search.BOx-Search.Result.Workign.with.Custom.List-Not.showing.ALLitems.View.Columns.pdf)

**Comment by wobba** [comment]
There's a small journey to be had to make this happen taking you from SharePoint column, to crawled property, to managed property 🙂

Take a look at https://docs.microsoft.com/en-us/microsoft-365/community/how-do-site-columns-become-managed-properties-thus-available-for-search and https://www.techmikael.com/2014/07/what-makes-sharepoint-column-searchable.html

**Comment by RamBalani** [comment]
Many thanks. I did come across this link you provided-now clear as ‘mud’!

Where I am confused is this—there is I thought a distinction between site columns and columns (regular) unique to a list.

When creating a new list -most of the time  I Export an Excel—seems Sharepoint automatically indexes the list items data.

That conclusion comes to me since the list becomes instantly search enabled without me setting up site columns or mapping a custom/list column to a managed property.

So when configuring PnP Search Results – though my PnP is able to search with keywords I enter in the PnP search box—only Title, Created and Summary columns are displayed in the search results.

I can’t seem o find a way to ‘force’ my other list columns to show up.

How do I cross the chasm here ?

**Reply by wobba** [reply]
You need to either use site columns which auto-generate managed properties with a naming convention, or you need to map the crawled property matching your column to e.g. RefinableString100. Once you know which properties to use, then you include them. There should be other posts out there explaining how to close this. It's the same procedure since SharePoint 2013.

**Reply by RamBalani** [reply]
Thanks.
That was known to us already, i.e EITHER:
 (1) use Site columns (Add new column from Existing Site Column) or 
((2) map a NEW regular List column where SP only generates Crawled property-so therefore we need to map that crawled property to a managed property MANUALLY

What I am asking to put  it another way - if I have an Excel file /table and I name the columns to match identical names of NOW  EXISTING SITE COLUMNS ALREADY on the SP site (Previously defined and existing ), will these Excel columns be identified by Sharepoint and correlated to the EXISTINS SITE COLUMNS already there) when I export the Excel file as NEW Sharepoint list?
That is the crux of the question.
The attachment here show our existing Site Columns with distinct names--if we name the Excel file columns to match the attachment column names then export to SP a NEW LIST --the question is -will Sharepoint recognize them as SITE COLUMNS ( as though they were create from 'Add from Existing Site Columns)
FOLLOW ?

![Showing Pre-existing Site Columns to ISO-1348 Group](https://user-images.githubusercontent.com/37209670/161540035-139460a2-2397-4c7e-88d4-6da01f406498.jpg)

**Reply by wobba** [reply]
No, the columns on the list has to be site columns for the auto mapping to occur.

**Comment by RamBalani** [comment]
I understand that - the question is will naming the Excel columns  in table in Excel to be exported as SP List if  identical to existing Sharepoint tenant Site Columns named do the job ? I do not want to create the list in Sharepoint at the get go  then have to write a Power Automate flow to read each Excel row then add to the list.
So once again- the data and column names identical to Site Columns will come from Excel, data rows exported to Sharepoint tenant site with Site Column same as site  names and all data from Excel.

**Reply by wobba** [reply]
You can test, but I don't think it will work. So you work around by mapping ows_colname crawled properties manually to RefinableXXYY managed properties instead - which will work.

**Reply by RamBalani** [reply]
Issue I have - with  "work around by mapping ows_colname crawled properties manually to RefinableXXYY managed properties instead "  --- mu Site Schema  shows ows_colname crawled properties ARE GREYED ! - 
so it seems not possible for me to MAP since ows_colname crawled properties are GREYED , i.e. Add Mapping click is NOT enabled--! Why is that ?

![Showing Ows_Colname for ISO-13485 Crawled Properties](https://user-images.githubusercontent.com/37209670/161767337-8ddff28a-080a-4f4d-8b35-3ee29f0cfd51.jpg)

![ADD Mapping Greyed for-- see  sample  Ows_Colname for ISO-13485 Crawled Properties-](https://user-images.githubusercontent.com/37209670/161767361-c249adb5-7b7e-40be-9e38-23cb7351b2b7.jpg)

**Reply by RamBalani** [reply]
We figured it out - buried in the documentation on Microsoft Schema, i.e. we can have edit access to map, create new managed property, etc on our tenant SITE COLLECTION level.
Thanks.

**Reply by wobba** [reply]
When mapping at a site, ensure you know what this means in terms of what you are solving. If you create a new managed property at a site, then this cannot be used when searching outside of that side - as it only lives in that one site. 

Also see my post https://www.techmikael.com/2016/02/search-schema-mapping-level-explained.html

**Reply by RamBalani** [reply]
Thanks for the info-but we're covered there - i.e intent for new managed property to that list is for ONE site only!
Appreciate   the post reference-will check out.

BTW Also  reading your FAST Search book you co-authored, interesting  FAST /Microsoft history, i.e. book available on my Safari Online when I searched 'Microsoft Search'

**Reply by wobba** [reply]
Happy to hear you find the book interesting, and I did author most of it as it was my project 🙂 Actually used it as reference myself today.

**Reply by RamBalani** [reply]
Going through your FAST book at snail's pace-we're newbies to Microsoft Search let alone you wonderful PnP Search!

Actually just paid a consultant who chose to use PnP plug-in to achieve what I just (myself 😏👍) completed today.

Been a struggle-but having gone through the process, as you put it earlier- I feel I've accomplished something-indeed has been a long journey from crawled to managed, then sidetracked with RefinableString00 and Tenant Level schema to bypass site greyed issue etc -feels like i just visited another planet in space but safely returned to Earth !

May have another question soon  then will join you for more coffee ☕ ☕ ☕ ☕ 

![Process NEAR management PnP Search Works Finally](https://user-images.githubusercontent.com/37209670/162257613-551717c0-5879-4df4-944c-f1597a74e68c.jpg)

**Reply by wobba** [reply]
I would be interested to know what documentation you would like to see and in what format for newcomers coming into search with SharePoint content.

**Reply by RamBalani** [reply]
Documentation needs to be CLEAR and COMPLETE and CURRENT!
Some behind the scenes goings-on by **Sharepoint search would be welcome** - these are mostly presumed to be understood by all and clearly they are not, something similar to   your FAST book background info peppered here and there  would be nice 👌 as an example.
 Documentation should explain (for dummies) the 'abc's of search schema concepts and procedures, crawling and indexing operations for LISTS vs Documents.

 I have only discovered, for instance, when exporting Excel as list into SP -the 1st column is never generated as a crawled property! WHY ? What happens BEHIND THE SCENES ? Lost a few strands of hair on that one! See attachment.

One big 'beef' -Microsoft does super-poorly on alerts and notifications as to what's been deprecated or amended-e.g. one can add Gallery View on List when viewing the List Columns UI   but not in List Settings as an example, also if I am right (am I?)-- Local Workbench testing has been deprecated if am not mistaken for SPFx testing. 

I have spun my wheels far too many times only to find out what I was attempting to do impossible since no longer supported - see attachment.
 
In other words-treat newbies as 'dummies' cause **_we really are not all Search experts no more than you are an FDA medical device regulatory expert as I am._**

[Shared with Svenson-Github Nice to Have for Search Dummies.docx](https://github.com/microsoft-search/pnp-modern-search/files/8453810/Shared.with.Svenson-Github.Nice.to.Have.for.Search.Dummies.docx)

**Reply by wobba** [reply]
@RamBalani CLEAR is very subjective, CURRENT is a moving target. COMPLETE is very dependent on what you aim to do.. because too much information makes simpler things hard to understand. And behind the scenes logic will likely not be shared, as that is internal Microsoft logic. But many concepts can indeed be shared to understand how content becomes searchable etc.

**Reply by RamBalani** [reply]
### Further to my point-and if I can BORROW from your very own bool I am reviewing!
See attachment.
**Now** - that's what I call a  **CLEAR and COMPLETE, CURRENT** also-since I assume that Microsoft has not changed how it CRAWLS a LIST once CREATED on Sharepoint since the time your book was published.

![Great Example of Completeness Mate-Your Own Book](https://user-images.githubusercontent.com/37209670/162719744-01277798-e4f1-433b-81b2-b1619cfef8f4.jpg)

**Comment by RamBalani** [comment]
@wobba  Good points all but disagree for reasons below:
**CLEAR** - is simple, i.e. for beginners--one assumes no previous SEARCH knowhow.
**CURRENT**- Microsoft keeps track of what is in PRODUCTION vs NO LONGER SUPPORTED-a matter of tracking and recording-then sharing public.
**COMPLETE**- also not ambiguous in my mind-e.g. when a list is uploaded that has 5 columns-only 4 Crawled properties are auto-created by Sharepoint? That's no Microsoft secret-i.e. discoverable when one searches the SCHEMA new CRAWLED PROPERTIES elements generated that correspond to the custom LIST  named columns, .e.g see attachment-so COMPLETE means a proper SEARCH book should explain what's happening there.
My 2-cents worth mate.
[5 Columns on my custom list below-4Crawls only.docx](https://github.com/microsoft-search/pnp-modern-search/files/8462691/5.Columns.on.my.custom.list.below-4Crawls.only.docx)

**Comment by iamhuman2414** [comment]
Hi @RamBalani, I'm a newbie, I have the same dilemma. Would you be able to give me detailed instructions on how you made your imported list (from excel) searchable and showing in the Pnp Search Results web part??  

Could you please elaborate? I'm not exactly sure how you did it.


> We figured it out - buried in the documentation on Microsoft Schema, i.e. we can have edit access to map, create new managed property, etc on our tenant SITE COLLECTION level.
> Thanks.

**Reply by iamhuman2414** [reply]
I only have two columns - Title and Technology. I searched for Technology column in the crawled properties but it's not there.

**Reply by RamBalani** [reply]
Did you add a list item after exporting your Excel file? Sharepoint for non-site columns will initiate the crawl only after adding data-weird but found out the hard way.
Add a list line item, give it a few minutes then search again for the 'Technology' crawled property SP creates.
Hope this helps.

**Reply by iamhuman2414** [reply]
My list contains data from Excel. I imported everything from my spreadsheet.

**Reply by iamhuman2414** [reply]
Also, I recreated the list by importing my excel spreadsheet with data, assigned a new list name, and added a test list item. waited for two hours, my 'Technology' column is still not listed in either managed or crawled properties

**Comment by RamBalani** [comment]
Add NEW list item on Sharepoint Online--not during export from Excel worksheet file. See what that does.
Also rename one of your Excel Column as distinct as possible, e.g. iamhuman2414Technology

**Reply by iamhuman2414** [reply]
Sorry for the confusion. I import, and not export, from Excel to Sharepoint by creating a new list. I renamed my column to something unique non-vocabulary word. 

I'm trying the option to create a blank list with just two columns (Title and 2414Tech), but I see no option to import the data from my Excel file.

**Comment by RamBalani** [comment]
Add new list **not by Importing or exporting** but by  menu + New List  menu item on your Modern List is what you need to do.
